scripts/processAndDownloadContracts.ts

```diff
@@ -17,21 +17,25 @@ async function processContracts() {
 			...await getAaveV2Misc()
 		]
 	const output = `
-
-const contractData: [string, string, string, string | null][] = ${JSON.stringify(contractData.map((x) => [addressString(x.address), x.data.name, x.data.protocol, x.data.logoURI]), null, 1)}
+import * as contractData from './contractMetadata.json';
 export type ContractDefinition = {
-    name: string,
-    logoURI: string | undefined,
-    protocol: string | undefined
+	name: string,
+	logoURI: string | undefined,
+	protocol: string | undefined
 }
 export const contractMetadata = new Map<string, ContractDefinition>(
-    contractData.map( ([address, name, protocol, logoURI] : [string, string, string, string | null]) => [address, {
-      name: name,
-      protocol: protocol,
-      logoURI: logoURI === null ? undefined : logoURI
-    }])
+	contractData.reduce(( acc, [address, name, protocol, logoURI] ) => {
+		if (!address || !name) return acc
+		return acc.concat([
+			[address, {
+				name: name,
+				protocol: protocol === null ? undefined : protocol,
+				logoURI: logoURI === null ? undefined : logoURI
+			}]])
```

Comment:
Consider push instead of concat here.  Also, I prefer first letter capitalized in acronyms so you avoid `XMLHTTPRPC` ridiculousness (preferring `XmlHttpRpc` instead).  Also consider missing instead of undefined.
```suggestion
		return acc.push([
			address,
			{
				name: name,
				...protocol ? {protocol} : {},
				...logoUri ? {logoUri} : {},
			}
		])
```

Reply:
Push doesn't work with reduce, it returns the length of the array